src/calculate

```diff
@@ -8,38 +8,22 @@ dbg() {
 }
 
 publish() {
-  msg=$(cat | head -n1)
-  jq -c -n '$ARGS.named' \
-    --argjson properties '{}' \
-    --arg routing_key "queue-publish" \
-    --arg payload_encoding "string" \
-    --arg payload "$(
-      jq -c -n '$ARGS.named' \
-        --arg target "#nixos" \
-        --arg body "$msg" \
-        --arg message_type "notice"
-    )" |
-    curl -v \
-      -H "content-type:application/json" \
-      -XPOST \
-      "$AMQPAPI"/api/exchanges/ircbot/amq.default/publish \
-      --data @-
+  msgplain=$(cat | head -n1)
+  msgformatted=$(cat | head -n1)
+
+  # Ensure that the bot is in the room.
+  curl -s -X POST \
+    "$SERVER/_matrix/client/r0/join/$ROOM_ID?access_token=$TOKEN" > /dev/null
```

Comment:
I don't like this. It's joining the room before sending every message. Not sure where to put it, though.

Reply:
Yes, we could use this and parse the JSON. https://matrix.org/docs/spec/client_server/r0.6.0#get-matrix-client-r0-joined-rooms

The more I work on this, the more I want to rewrite in something that isn't bash haha

Reply:
Please do! Or at least the parts of it that you need to!